app/models/survey_question.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class SurveyQuestion < ActiveRecord::Base
+  belongs_to :survey
+  has_many :survey_replies, dependent: :destroy
+
+  # Order of this list should not be changed without proper action!
+  enum kind: [:boolean, :choice, :string, :text, :datetime, :numeric]
+
+  ICONS = { boolean: 'dot-circle-o', choice: 'check-square-o', string: 'edit', text: 'align-left', datetime: 'clock-o', numeric: 'slack' }.freeze
+
+  validates :title, presence: true
+  validates :possible_answers, :max_choices, :min_choices, presence: true, if: 'choice?'
+  validates :min_choices, numericality: { greater_than_or_equal_to: 1 }, allow_blank: true
+  validates :max_choices, numericality: { greater_than_or_equal_to: 1 }, allow_blank: true
+
+  validate :max_choices_greater_than_min
+
+  def single_choice?
+    choice? && max_choices == 1 && min_choices == 1
+  end
+
+  def multiple_choice?
+    choice? && max_choices > 1
+  end
+
+  def possible_answers=(value)
+    self[:possible_answers] = choice? ? value : nil
```

Comment:
is the `nil` needed here? :thinking:

Reply:
I think that's just to make it obvious the value is going to be nil unless this is a question of choice kind.

Reply:
but we can write this like:

`self[:possible_answers] = value if choice?`

it also happens in `min_choices` an `min_choices`

anyway, I would be happy if it is fixed, but this is also not that important, so do not block merging because of that.